chain-impl-mockchain/src/block.rs

```diff
@@ -69,6 +70,80 @@ impl property::HasTransaction<SignedTransaction> for Block {
     }
 }
 
+/// `Block` is an element of the blockchain it contains multiple
+/// transaction and a reference to the parent block. Alongside
+/// with the position of that block in the chain.
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+pub struct SignedBlock {
+    /// Internal block.
+    block: Block,
+    /// Public key used to sign the block.
+    public_key: PublicKey,
+    /// List of cryptographic signatures that verifies the block.
+    signature: Signature,
+}
+
+impl SignedBlock {
+    /// Create a new signed block.
+    pub fn new(block: Block, pkey: PrivateKey) -> Self {
+        use chain_core::property::Block;
+        let block_id = block.id();
+        SignedBlock {
+            block: block,
+            public_key: pkey.public(),
+            signature: pkey.sign(block_id.as_ref()),
+        }
+    }
+
+    /// Verify if block is correctly signed by the key.
+    /// Return `false` if there is no such signature or
+    /// if it can't be verified.
+    pub fn verify(&self) -> bool {
+        use chain_core::property::Block;
+        let block_id = self.block.id();
+        self.public_key.verify(block_id.as_ref(), &self.signature)
+    }
+}
+
+impl property::Serializable for SignedBlock {
+    type Error = bincode::Error;
+
+    fn deserialize<R: std::io::Read>(reader: R) -> Result<Self, bincode::Error> {
+        bincode::deserialize_from(reader)
+    }
+
+    fn serialize<W: std::io::Write>(&self, writer: W) -> Result<(), Self::Error> {
+        bincode::serialize_into(writer, self)
+    }
+}
+
+impl property::Block for SignedBlock {
+    type Id = <Block as property::Block>::Id;
+    type Date = <Block as property::Block>::Date;
+    type Header = <Block as property::Block>::Header;
+
+    /// Identifier of the block, currently the hash of the
+    /// serialized transaction.
+    fn id(&self) -> Self::Id {
+        self.block.id()
+    }
+
+
```

Comment:
rustfmt is going to fail on this line. please remove.

Reply:
Fixed